[#80] Add upload progress bar for file upload

gabemart opened the pull request:
#80

jonase commented:
@gabemart the progress bar is really nice! I wonder if we should leave it visible at 100% when the upload is finished instead of hide it?

gabemart commented:
@jonase I have tested out your suggestion and I like it 👍

jonase commented on client/src/components/modals/createDataset/DataFileDataSourceSettings.jsx:
```diff
@@ -35,17 +42,23 @@ export default class DataFileDataSourceSettings extends Component {
       endpoint: '/api/files',
       onError(error) {
         console.error(`Failed because: ${error}`);
+        handleProgress(null);
```
In `handleProgress` we `parseFloat` the `percentage` argument and `parseFloat(null)` evaluates to `NaN` which I don't think is desired here, especially since we explicitly check for `null` in `getProgressBarBool`.

jonase commented on client/src/components/modals/createDataset/DataFileDataSourceSettings.jsx:
```diff
@@ -17,6 +18,12 @@ export default class DataFileDataSourceSettings extends Component {
+    this.state = { uploadProgressPercentage: null };
+  }
+
+  getShowProgressBarBool() {
```
I would suggest renaming this method to remove the type suffix. Perhaps simply `showProgressBar`? or `isProgressBarVisible`? I don't think we do much data type suffix/prefix based naming currently and I like it that way.

gabemart commented:
@jonase I agree with both your comments and have made some changes.

* I renamed the function to `isProgressBarVisible()` as suggested

* After revisiting error-handling to correct the problem with parsing null, I added some optional text props to the progress bar component so it will display to the user if there is an error and show a "success" message if the upload completes

jonase commented:
👍
